ciao-controller/controller_test.go

```diff
@@ -742,7 +723,12 @@ func testStartTracedWorkload(t *testing.T) *testutil.SsntpTestClient {
 		t.Fatal(err)
 	}
 
-	client := newTestClient(0, ssntp.AGENT)
+	client, err := testutil.NewSsntpTestClientConnection("StartTracedWorkload", ssntp.AGENT)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// deliberately leak
+	//defer client.Ssntp.Close()
 
```

Comment:
It looks to me like you're deliberately not closing the connection.  But I'm inferring the design of the overall test.  You tell me :)